### ISSUE TITLE
Creating a REST API for LLM-VM 

### ISSUE THREAD

**Issue by zachsueh2023** [anarchy-ai/LLM-VM#250]

**Comment by mmirman** [comment]
Maybe create an api key provisioning system too?

**Comment by zachsueh2023** [comment]
Yes, we need the api key and the user needs to be registered. Let the user apply for a trial first, and we'll review it.

**Comment by bilal-aamer** [comment]
Do we have a user registration already?

**Comment by WadeFC** [comment]
i dont think so

**Comment by zachsueh2023** [comment]
Yeah, thank you for your feedback, we love to hear any feedback and discussion.

**Comment by mmirman** [comment]
@zachsueh2023 True that!  Thanks for helping teach the anarchy way :-)

**Comment by jaywha** [comment]
I've done some API management in Azure before and even did some "midair" transformations for authentication using Azure API Management tools (I think for HMAC but can't quite remember). I like working with APIs in general anyway so I can conform to any requirements as needed or do some research as to what are the best practices. I can also look into AWS or Google Cloud, if preferred.

**Comment by zhsueh** [comment]
@jaywha I'm not sure that the anarchy will use Azure, Google and AWS cloud, but you can look at rest api papers on google scholar. Yes, you can also go ahead and research these clouds so you can better contribute when anarchy decides which one to use.

**Comment by MananDoshi1301** [comment]
@mmirman I'd love to collaborate if someone is already working on this topic!

**Comment by Jobhdez** [comment]
@VictorOdede what would the endpoints be? which language model call? I have done rest apis in djangorestframework

**Comment by VictorOdede** [comment]
@Jobhdez @MananDoshi1301 @jaywha  will add a few tasks around this by tomorrow. Feel free to pick them up.
 if you're interested.

**Comment by MananDoshi1301** [comment]
@VictorOdede Yes sure Vik, thanks. Would be very happy to contribute!

**Comment by jaywha** [comment]
Same here! We can divide and conquer this issue 😁

**Comment by lucylililiwang** [comment]
Hi, can I please work on this issue? Thank you!

**Comment by lucylililiwang** [comment]
https://github.com/lucylililiwang/LLM-VM/blob/lucylililiwangfrom_rest-api/rest_api.py
